dotCMS/src/integration-test/java/com/dotcms/util/IntegrationTestInitService.java

```diff
@@ -48,10 +55,61 @@ public void init() throws Exception {
             // Init other dotCMS services.
             DotInitializationService.getInstance().initialize();
 
+            if (Config.getBooleanProperty("dotcms.integrationtest.run.upgradetask", true)) {
+                this.tryUpgradeTask();
+            }
+
             initCompleted.set(true);
         }
     }
-    
+
+    private void tryUpgradeTask () {
+
+        final StartupAPI startupAPI = APILocator.getStartupAPI();
+
+        final List<Class<?>> alwaysTaskClasses = startupAPI.getStartupRunAlwaysTaskClasses();
+        final List<Class<?>> onceTaskClasses   = startupAPI.getStartupRunOnceTaskClasses();
+
+        Logger.info(this, ()->"Running the always upgrade tasks");
+        for (final Class<?> alwaysTaskClass : alwaysTaskClasses) {
+
+            try {
+
+                Logger.info(this, ()->"Running the upgrade task: " + alwaysTaskClass.getCanonicalName());
+                startupAPI.runStartup(alwaysTaskClass);
+            } catch (DotDataException e) {
+                Logger.info(this, ()->"Error on running the upgrade task: " +
+                        alwaysTaskClass.getCanonicalName() + ", msg: " + e.getMessage());
+
+                if (Config.getBooleanProperty("dotcms.integrationtest.run.upgradetask.stoponerror", true)) {
+                    fail(e.getMessage());
+                }
+            }
+        }
+        Logger.info(this, ()->"Ran the always upgrade tasks");
```

Comment:
why do you use two consecutive logs with different messages?

Reply:
there are two process, the first one is about the end of the first process, and the second log is about the beginning of the second one!